nimare/io.py

```diff
@@ -275,3 +278,123 @@ def convert_sleuth_to_dataset(text_file, target="ale_2mm"):
+        be in neurovault and the values are the name of the respective
+        statistic map in a nimare dataset. Default = None.
+    **dset_kwargs : keyword arguments passed to Dataset
+        Keyword arguments to pass in when creating the Dataset object.
+        see :obj:`nimare.dataset.Dataset` for details.
+
+    Returns
+    -------
+    :obj:`nimare.dataset.Dataset`
+        Dataset object containing experiment information from neurovault.
+    """
+    if img_dir is None:
+        img_dir = Path(tempfile.mkdtemp(prefix="neurovault_"))
+    else:
+        img_dir = Path(img_dir, exist_ok=True)
+        img_dir.mkdir(parents=True, exist_ok=True)
+
+    if map_type_conversion is None:
+        map_type_conversion = {
+            "T map": "t",
+            "variance": "varcope",
+            "univariate-beta map": "beta",
+        }
+
+    if not isinstance(collection_ids, dict):
+        collection_ids = {nv_coll: nv_coll for nv_coll in collection_ids}
+
+    dataset_dict = {}
+    for coll_name, nv_coll in collection_ids.items():
+        for contrast_name, contrast_ids in contrasts.items():
+            nv_url = f"https://neurovault.org/api/collections/{nv_coll}/images/?format=json"
+            images = requests.get(nv_url).json()
+
+            dataset_dict[f"study-{coll_name}"] = {
+                "contrasts": {
+                    contrast_name: {
+                        "images": {
+                            "beta": None,
+                            "t": None,
+                            "varcope": None,
+                        },
+                        "metadata": {"sample_sizes": None},
+                    }
+                }
+            }
+
+            for img_dict in images["results"]:
+                if not (
+                    (
+                        img_dict["name"] in contrast_ids
+                        or any(
+                            [
+                                img_dict["name"].startswith(f"{contrast}_stat")
+                                for contrast in contrast_ids
+                            ]
+                        )
+                    )
+                    and img_dict["map_type"] in map_type_conversion
+                ):
+                    continue
+
+                filename = img_dir / Path(img_dict["file"]).name
+
+                if not filename.exists():
+                    r = requests.get(img_dict["file"])
+                    with open(filename, "wb") as f:
+                        f.write(r.content)
+
+                (
+                    dataset_dict[f"study-{coll_name}"]["contrasts"][contrast_name]["images"][
+                        map_type_conversion[img_dict["map_type"]]
+                    ]
+                ) = filename.as_posix()
+
+                # temporary sample size
+                sample_size = img_dict["number_of_subjects"]
+
+            # take the sample size from the final valid image entry (they should all be the same)
```

Comment:
Hmm... It's true that the sample size *should* be the same, but we can't guarantee that. They could be different if, e.g., the contrast matching procedure went awry, or if there was a data entry problem. Probably this is something we should warn the user about. Not sure what the best way to implement that is—and maybe it should be handled inside the `Dataset` initializer rather than here.

Reply:
added some logic here to make sure sample_size is the same across images, and chooses most common sample_size if there are multiple options.